driver-scala/src/main/scala/org/mongodb/scala/package.scala

```diff
@@ -392,9 +399,18 @@ package object scala extends ClientSessionImplicits with ObservableImplicits wit
 
   implicit def bsonDocumentToDocument(doc: BsonDocument): Document = new Document(doc)
 
-  implicit def bsonDocumentToUntypedDocument(doc: BsonDocument): org.bson.Document =
-    org.bson.Document.parse(doc.toJson())
+  implicit def documentToUntypedDocument(doc: Document): org.bson.Document =
+    bsonDocumentToUntypedDocument(doc.underlying)
+
+  implicit def bsonDocumentToUntypedDocument(doc: BsonDocument): org.bson.Document = {
+    val writer: BsonBinaryWriter = new BsonBinaryWriter(new BasicOutputBuffer())
```

Comment:
You can use BsonDocumentReader and simplify to

    new DocumentCodec().decode(new BsonDocumentReader(doc), DecoderContext.builder().build())

Reply:
Much nicer thanks